library/src/androidMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt

```diff
@@ -0,0 +1,75 @@
+package com.badoo.reaktive.scheduler
+
+import android.os.Handler
+import android.os.Looper
+import com.badoo.reaktive.disposable.CompositeDisposable
+
+internal class MainScheduler : Scheduler {
+
+    private val disposables = CompositeDisposable()
+
+    override fun newExecutor(): Scheduler.Executor =
+        ExecutorImpl()
+            .also(disposables::add)
+
+    override fun destroy() {
+        disposables.dispose()
+    }
+
+    private class ExecutorImpl : Scheduler.Executor {
+        @Volatile
```

Comment:
Do we need Volatile here?

Reply:
Absolutely, as it's accessed from isDisposed's get method where there is no synchronisation.